src/reflection.jl

```diff
@@ -134,16 +136,17 @@ function process_info(interp, @nospecialize(info), argtypes::ArgTypes, @nospecia
                 infos[i]
             elseif (@static isdefined(Compiler, :ConstResult) && true) && isa(result, Compiler.ConstResult)
                 linfo = result.mi
-                ConstEvalCallInfo(MICallInfo(linfo, rt), argtypes)
+                mici = MICallInfo(linfo, rt, haskey(interp.unopt, linfo) ? interp.unopt[linfo].effects : Effects())
+                ConstEvalCallInfo(mici, argtypes)
             else
                 @assert isa(result, Compiler.InferenceResult)
                 linfo = result.linfo
-                mici = MICallInfo(linfo, rt)
+                mici = MICallInfo(linfo, rt, result.ipo_effects)
```

Comment:
This needs to be conditional on the Julia version

Reply:
👍  I'll refactor a bit to make this backwards compatible. Thanks for the review(s)!